### PR TITLE
#1689 Update Comp & Pen notification filters

### PR DESCRIPTION
**Pull request by kalbfled**
# Description

The API contains a Celery Beat task the query DynamoDB for information to send a Comp & Pen notification.  We should only send notifications for "items" (no-sql term; like a row in sql) for which:
1. is_processed is False
2. has_duplicate_mappings is not set or False
3. payment_id is not equal to -1
4. paymentAmount exists

These changes add filters 2, 3, and 4 (above), update outdated information in tests/README.md, enable comp & pen in Perf, and correct a typo in task definition variable names.

issue #1689

## Type of change

Please check the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

@cris-oddball and I collaborated during the development process.  Please see her notes in the ticket.

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column

### CODE REVIEW

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: app/celery/scheduled_tasks.py [hunk 1]
```diff
@@ -347,7 +358,7 @@ def send_scheduled_comp_and_pen_sms():
                 ReturnValues='ALL_NEW',
             )
 
-            current_app.logger.info('updated_item from dynamodb ("is_processed" shouldb be "True"): %s', updated_item)
+            current_app.logger.info('updated_item from dynamodb ("is_processed" should be "True"): %s', updated_item)
```
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/celery/scheduled_tasks.py]: changed

File: app/celery/scheduled_tasks.py [hunk 2]
```diff
@@ -206,15 +207,25 @@ def _get_dynamodb_comp_pen_messages(
...
     :param message_limit: the number of rows to search at a time and the max number of items that should be returned
     :return: a list of entries from the table that have not been processed yet
+
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/dynamodb.html#querying-and-scanning
     """
 
-    results = table.scan(FilterExpression=boto3.dynamodb.conditions.Attr('is_processed').eq(False), Limit=message_limit)
+    filters = Attr('is_processed').not_exists() | Attr('is_processed').eq(False)
+    filters = filters & Attr('payment_id').exists() & Attr('payment_id').ne(-1) & Attr('paymentAmount').exists()
+    filters = filters & (Attr('has_duplicate_mappings').not_exists() | Attr('has_duplicate_mappings').eq(False))
```
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/celery/scheduled_tasks.py]: Revised similarly to this.

File: app/celery/scheduled_tasks.py [hunk 3]
```diff
@@ -206,15 +207,29 @@ def _get_dynamodb_comp_pen_messages(
...
     :return: a list of entries from the table that have not been processed yet
+
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/dynamodb.html#querying-and-scanning
     """
 
-    results = table.scan(FilterExpression=boto3.dynamodb.conditions.Attr('is_processed').eq(False), Limit=message_limit)
+    filters = (
+        Attr('is_processed').eq(False)
+        & Attr('payment_id').exists()
+        & Attr('payment_id').ne(-1)
```
Comment (MackHalliday) [inline app/celery/scheduled_tasks.py:227]: Should the value of `payment_id` be a constant? I saw we used a constant in the Consumer repo.

**Review by MackHalliday — APPROVED** [review]
Reviewed new logic and testing - left nits. Approved.

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**